docs/5.0/kubernetes-5.0-migration.md

```diff
@@ -0,0 +1,392 @@
+---
+title: Kubernetes 5.0 Migration Guide
+description: How to migrate pre-5.0 Teleport clusters to new configuration
+---
+
+# Kubernetes intergartion in 5.0
+
+In release 5.0, Teleport has changed the [Kubernetes
+integration](kubernetes-ssh.md) to improve configuration and user experience.
+These changes are backwards compatible and there is no _required_ migration.
+However, to get the most out of Teleport, you should consider migrating as
+described below.
+
+The main changes in 5.0 are:
+- new `kubernetes_service` configuration section in `teleport.yaml`, decoupled
+  from `proxy_service`
+- support for multiple Kubernetes clusters per Teleport cluster (replaces the
+  need for [Trusted Clusters](admin-guide.md#trusted-clusters))
+- RBAC support for Kubernetes clusters
+
+Important note: `proxy_service` with an open port for Kubernetes requests is
+still required. A proxy is always the public-facing gateway into the Teleport
+cluster, acting as an authentication point and a connection router.
+
+# Example scenarios
+
+We'll describe the migration for several common Teleport configurations.
+
+## Standalone proxy for a single cluster
+
+In this scenario, we're running a Teleport proxy outside of the Kubernetes
+cluster and connecting to it using a `kubeconfig_file`.
+
+Before 5.0:
+
+```yaml
+# teleport.yaml
+proxy_service:
+  kubernetes:
+    enabled: yes
+    listen_addr: 0.0.0.0:3026
+    kubeconfig_file: /path/to/kubeconfig
+```
+
+After 5.0:
+
+```yaml
+# teleport.yaml
+proxy_service:
+  kube_listen_addr: 0.0.0.0:3026
+
+kubernetes_service:
+  enabled: yes
+  listen_addr: 0.0.0.0:3027
+  kubeconfig_file: /path/to/kubeconfig
+```
+
+The `listen_port` for `kubernetes_service` only needs to be exposed to the
+Proxies within your cluster. This port will not be directly used by the
+clients.
+
+## Proxy in a Kubernetes pod of a single cluster
+
+In this scenario, we're running a Teleport proxy inside of the Kubernetes
+cluster and connecting to it using the pod service account credentials.
+
+Before 5.0:
+
+```yaml
+# teleport.yaml
+proxy_service:
+  kubernetes:
+    enabled: yes
+    listen_addr: 0.0.0.0:3026
+```
+
+After 5.0:
+
+```yaml
+# teleport.yaml
+proxy_service:
+  kube_listen_addr: 0.0.0.0:3026
+
+kubernetes_service:
+  enabled: yes
+  listen_addr: 0.0.0.0:3027
+```
+
+## Multiple Kubernetes clusters
+
+In this scenario, we're running multiple Trusted Teleport clusters, one for
+each Kubernetes cluster.
+
+### Option 1: Teleport Kubernetes agents in each Kubernetes cluster
+
+This is the recommended approach for handling multiple Kubernetes clusters.
+
+Instead of running multiple Teleport clusters (auth and proxy for each
+Kubernetes cluster), we will connect multiple `kubernetes_service` agents to
+the same root Teleport cluster.
```

Comment:
Maybe Add some extra clarity about the fact that the connections are tunnelled?

```suggestion
Instead of running multiple Teleport clusters (auth and proxy for each
Kubernetes cluster), we will connect multiple `kubernetes_service` agents to
the same root Teleport cluster.

Each `kubernetes_service` will create its own reverse tunnel back to a Teleport
proxy to join the cluster.
```